colin/core/target.py

```diff
@@ -32,44 +32,51 @@
 
 class Target(object):
 
-    def __init__(self, name, logging_level):
-        self.name = name
-        self.instance = Target._get_target_instance(name, logging_level=logging_level)
+    def __init__(self, target, logging_level):
+        self.instance = Target._get_target_instance(target, logging_level=logging_level)
 
     @staticmethod
-    def _get_target_instance(target_name, logging_level):
+    def _get_target_instance(target, logging_level):
         """
         Get the Container/Image instance for the given name.
         (Container is the first choice.)
 
-        :param target_name: str
+        :param target: str or instance of Image/Container
         :return: Container/Image
         """
-        logger.debug("Finding target '{}'.".format(target_name))
+        logger.debug("Finding target '{}'.".format(target))
+
+        if isinstance(target, Image):
+            return target
+        if isinstance(target, Container):
+            return target
 
         with DockerBackend(logging_level=logging_level) as backend:
 
             try:
                 cont = backend.ContainerClass(image=None,
-                                              container_id=target_name)
+                                              container_id=target)
                 logger.debug("Target is a container.")
                 return cont
             except NotFound:
-                name_split = target_name.split(':')
-                if len(name_split) == 2:
-                    name, tag = name_split
-                    image = backend.ImageClass(repository=name,
-                                               tag=tag,
+
+                image_name = ImageName.parse(target)
+                logger.debug(str(image_name))
+
+                logger.debug("Finding image '{}' with tag '{}'.".format(image_name.name, image_name.tag))
```

Comment:
Do we need both debug messages? The above one looks like a leftover to me.

Reply:
Good point. Thanks.